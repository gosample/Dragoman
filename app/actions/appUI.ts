import * as actionHelpers from './actions';

const SET_SETTINGS_OPEN = 'SET_SETTINGS_OPEN';
const SET_ERROR_DIALOG_VISIBLE = 'SET_ERROR_DIALOG_VISIBLE';
const SET_ERROR_DIALOG_TITLE = 'SET_ERROR_DIALOG_TITLE';
const SET_ERROR_DIALOG_EXPLANATION = 'SET_ERROR_DIALOG_EXPLANATION';

export const setSettingsOpen = actionHelpers.actionCreator<boolean>(SET_SETTINGS_OPEN);
export const setErrorDialogVisible = actionHelpers.actionCreator<boolean>(SET_ERROR_DIALOG_VISIBLE);
export const setErrorDialogTitle = actionHelpers.actionCreator<string>(SET_ERROR_DIALOG_TITLE);
export const setErrorDialogExplanation = actionHelpers.actionCreator<string>(SET_ERROR_DIALOG_EXPLANATION);
