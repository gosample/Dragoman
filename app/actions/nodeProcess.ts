import * as actionHelpers from './actions';

const ADD_NODE_PROCESS_PID = 'ADD_NODE_PROCESS_PID';
const REMOVE_NODE_PROCESS_PID = 'REMOVE_NODE_PROCESS_BY_PID';
const CLEAR_NODE_PROCESSES_PIDS = 'CLEAR_NODE_PROCESSES_PIDS';

export const addNodeProcessPid = actionHelpers.actionCreator<number>(ADD_NODE_PROCESS_PID);
export const removeNodeProcessPid = actionHelpers.actionCreator<number>(REMOVE_NODE_PROCESS_PID);
export const clearNodeProcessPids = actionHelpers.emptyActionCreator(CLEAR_NODE_PROCESSES_PIDS);
