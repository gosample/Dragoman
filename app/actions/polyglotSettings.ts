import * as actionHelpers from './actions';

const CHANGE_PROTO_DISCOVERY_ROOT = 'CHANGE_PROTO_DISCOVERY_ROOT';
const CHANGE_ENDPOINT = 'CHANGE_ENDPOINT';
const CHANGE_CONFIG_SET_PATH = 'CHANGE_CONFIG_SET_PATH';
const CHANGE_CONFIG_NAME = 'CHANGE_CONFIG_NAME';
const CHANGE_TLS_CA_CERT_PATH = 'CHANGE_TLS_CA_CERT_PATH';
const CHANGE_DEADLINE_MS = 'CHANGE_DEADLINE_MS';
const CHANGE_ADD_PROTOC_INCLUDES = 'CHANGE_ADD_PROTOC_INCLUDES';

export const changeProtoDiscoveryRoot = actionHelpers.actionCreator<string>(CHANGE_PROTO_DISCOVERY_ROOT);
export const changeEndpoint = actionHelpers.actionCreator<string>(CHANGE_ENDPOINT);
export const changeConfigSetPath = actionHelpers.actionCreator<string>(CHANGE_CONFIG_SET_PATH);
export const changeConfigName = actionHelpers.actionCreator<string>(CHANGE_CONFIG_NAME);
export const changeTlsCaCertPath = actionHelpers.actionCreator<string>(CHANGE_TLS_CA_CERT_PATH);
export const changeDeadlineMs = actionHelpers.actionCreator<number>(CHANGE_DEADLINE_MS);
export const changeAddProtocIncludes = actionHelpers.actionCreator<string>(CHANGE_ADD_PROTOC_INCLUDES);

