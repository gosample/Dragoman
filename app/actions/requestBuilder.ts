import * as actionHelpers from './actions';

const SET_REQUEST = 'SET_REQUEST';
const SET_FULL_METHOD = 'SET_FULL_METHOD';
const SET_CALL_REQUEST_IN_PROGRESS = 'SET_CALL_REQUEST_IN_PROGRESS';
const SET_CLIENT_STREAMING_REQUEST = 'SET_CLIENT_STREAMING_REQUEST';

export const setRequest = actionHelpers.actionCreator<string>(SET_REQUEST);
export const setFullMethod = actionHelpers.actionCreator<string>(SET_FULL_METHOD);
export const setCallRequestInProgress = actionHelpers.actionCreator<boolean>(SET_CALL_REQUEST_IN_PROGRESS);
export const setClientStreamingRequest = actionHelpers.actionCreator<boolean>(SET_CLIENT_STREAMING_REQUEST);
