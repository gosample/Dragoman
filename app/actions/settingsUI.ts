import * as actionHelpers from './actions';

const SET_ENDPOINT_REQUIRED = 'SET_ENDPOINT_REQUIRED';
const SET_ENDPOINT_ERROR = 'SET_ENDPOINT_ERROR';
const SET_PROTO_DISCOVERY_ROOT_ERROR = 'SET_PROTO_DISCOVERY_ROOT_ERROR';
const SET_CONFIG_SET_PATH_ERROR = 'SET_CONFIG_SET_PATH_ERROR';
const SET_TLS_CA_CERT_PATH_ERROR = 'SET_TLS_CA_CERT_PATH_ERROR';
const SET_ADD_PROTOC_INCLUDES_ERRORS = 'SET_ADD_PROTOC_INCLUDES_ERRORS';
const SET_OAUTH_REFRESH_TOKEN_ENDPOINT_URL_ERROR = 'SET_OAUTH_REFRESH_TOKEN_ENDPOINT_URL_ERROR';
const SET_OAUTH_REFRESH_TOKEN_PATH_ERROR = 'SET_OAUTH_REFRESH_TOKEN_PATH_ERROR';
const SET_OAUTH_ACCESS_TOKEN_PATH_ERROR = 'SET_OAUTH_ACCESS_TOKEN_PATH_ERROR';

export const setEndpointRequired = actionHelpers.actionCreator<boolean>(SET_ENDPOINT_REQUIRED);
export const setEndpointError = actionHelpers.actionCreator<boolean>(SET_ENDPOINT_ERROR);
export const setProtoDiscoveryRootError = actionHelpers.actionCreator<boolean>(SET_PROTO_DISCOVERY_ROOT_ERROR);
export const setConfigSetPathError = actionHelpers.actionCreator<boolean>(SET_CONFIG_SET_PATH_ERROR);
export const setTlsCaCertPathError = actionHelpers.actionCreator<boolean>(SET_TLS_CA_CERT_PATH_ERROR);
export const setAddProtocIncludesError = actionHelpers.actionCreator<boolean[]>(SET_ADD_PROTOC_INCLUDES_ERRORS);
export const setOauthRefreshTokenEndpointUrlError = actionHelpers.actionCreator<boolean>(SET_OAUTH_REFRESH_TOKEN_ENDPOINT_URL_ERROR);
export const setOauthRefreshTokenPathError = actionHelpers.actionCreator<boolean>(SET_OAUTH_REFRESH_TOKEN_PATH_ERROR);
export const setOauthAccessTokenPathError = actionHelpers.actionCreator<boolean>(SET_OAUTH_ACCESS_TOKEN_PATH_ERROR);
