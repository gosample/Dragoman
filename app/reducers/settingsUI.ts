import { isActionOfType, Action } from '../actions/actions';
import * as SettingsUIActions from '../actions/settingsUI';

export type SettingsUIState = Readonly<{
  endpointRequired: boolean;
  endpointError: boolean;
  protoDiscoveryRootError: boolean;
  configSetPathError: boolean;
  addProtocIncludesErrors: boolean[];
  oauthRefreshTokenEndpointUrlError: boolean;
  oauthRefreshTokenPathError: boolean;
  oauthAccessTokenPathError: boolean;
  tlsCaCertPathError: boolean;
  tlsClientKeyPathError: boolean;
  tlsClientCertPathError: boolean;
  tlsClientOverrideAuthorityError: boolean;
}>;

const initialSettingsUIState: SettingsUIState = {
  endpointRequired: false,
  endpointError: false,
  protoDiscoveryRootError: false,
  configSetPathError: false,
  addProtocIncludesErrors: [],
  oauthRefreshTokenEndpointUrlError: false,
  oauthRefreshTokenPathError: false,
  oauthAccessTokenPathError: false,
  tlsCaCertPathError: false,
  tlsClientKeyPathError: false,
  tlsClientCertPathError: false,
  tlsClientOverrideAuthorityError: false,
};

export default function settingsUIReducer(state: SettingsUIState = initialSettingsUIState, action: Action<any>): SettingsUIState {

  if (isActionOfType(action, SettingsUIActions.setEndpointRequired)) {
    return {
      ...state,
      endpointRequired: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setEndpointError)) {
    return {
      ...state,
      endpointError: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setProtoDiscoveryRootError)) {
    return {
      ...state,
      protoDiscoveryRootError: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setConfigSetPathError)) {
    return {
      ...state,
      configSetPathError: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setAddProtocIncludesError)) {
    return {
      ...state,
      addProtocIncludesErrors: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setOauthRefreshTokenEndpointUrlError)) {
    return {
      ...state,
      oauthRefreshTokenEndpointUrlError: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setOauthRefreshTokenPathError)) {
    return {
      ...state,
      oauthRefreshTokenPathError: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setOauthAccessTokenPathError)) {
    return {
      ...state,
      oauthAccessTokenPathError: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setTlsCaCertPathError)) {
    return {
      ...state,
      tlsCaCertPathError: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setTlsClientCertPathError)) {
    return {
      ...state,
      tlsClientCertPathError: action.payload,
    };
  }
  if (isActionOfType(action, SettingsUIActions.setTlsClientKeyPathError)) {
    return {
      ...state,
      tlsClientKeyPathError: action.payload,
    };
  }
  return state;
}
