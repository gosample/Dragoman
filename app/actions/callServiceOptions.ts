import * as actionHelpers from './actions';
import { CallServiceOptions } from '../reducers/callServiceOptions';

const CHANGE_JSON_BODY = 'CHANGE_JSON_BODY';
const CHANGE_FULL_METHOD = 'CHANGE_FULL_METHOD';
const SET_CALL_SERVICE_OPTIONS = 'SET_CALL_SERVICE_OPTIONS';

export const setCallServiceOptions = actionHelpers.actionCreator<CallServiceOptions>(SET_CALL_SERVICE_OPTIONS);
export const changeJsonBody = actionHelpers.actionCreator<string>(CHANGE_JSON_BODY);
export const changeFullMethod = actionHelpers.actionCreator<string>(CHANGE_FULL_METHOD);
