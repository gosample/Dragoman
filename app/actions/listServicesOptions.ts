import * as actionHelpers from './actions';
import { ListServicesOptions } from '../reducers/listServicesOptions';

const SET_LIST_SERVICES_OPTIONS = 'ListServicesOptions';
const CHANGE_SERVICE_FILTER = 'CHANGE_JSON_BODY';
const CHANGE_METHOD_FILTER = 'CHANGE_FULL_METHOD';

export const setListServicesOptions = actionHelpers.actionCreator<ListServicesOptions>(SET_LIST_SERVICES_OPTIONS);
export const changeServiceFilter = actionHelpers.actionCreator<string>(CHANGE_SERVICE_FILTER);
export const changeMethodFilter = actionHelpers.actionCreator<string>(CHANGE_METHOD_FILTER);
