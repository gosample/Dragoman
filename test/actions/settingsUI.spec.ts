import * as settingsUIActions from '../../app/actions/settingsUI';

describe('settings UI actions', () => {
  it('setEndpointError should create set endpoint error action', () => {
    expect(settingsUIActions.setEndpointError(false)).toMatchSnapshot();
  });
});
