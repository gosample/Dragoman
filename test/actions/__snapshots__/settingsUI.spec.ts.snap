// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings UI actions setEndpointError should create set endpoint error action 1`] = `
Object {
  "error": false,
  "payload": false,
  "type": "SET_ENDPOINT_ERROR",
}
`;
